src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/activity/LanternMainActivity.java

```diff
@@ -104,57 +182,289 @@ protected void onCreate(Bundle savedInstanceState) {
+      
+        // store show/hide feed preference
+        session.updateFeedPreference(showFeed);
+        showFeedview();
+
+        if (menuOptionIndex >= 0 && menuOptionIndex < mNavItems.size()) {
+            if (showFeed) {
+                mNavItems.get(menuOptionIndex).setTitle(resources.getString(R.string.newsfeed_off_option));
+            } else {
+                mNavItems.get(menuOptionIndex).setTitle(resources.getString(R.string.newsfeed_option));
+            }
         }
+        listAdapter.notifyDataSetChanged();
     }
 
-    // startLocalProxy starts a separate instance of Lantern
-    // used for proxying requests we need to make even before
-    // the user enables full-device VPN mode
-    private String startLocalProxy() {
-        // if the Lantern VPN is already running
-        // then we just fetch the feed without
-        // starting another local proxy
-        if (Service.isRunning(context)) {
-            return "";
+    private void updateStatus(boolean useVpn) {
+        displayStatus(useVpn);
+        changeFeedHeaderColor(useVpn);
+        session.updateVpnPreference(useVpn);
+    }
+
+    @Click(R.id.powerLantern)
+    public void switchLantern(View view) {
+
+        boolean on = ((ToggleButton)view).isChecked();
+
+        if (!Utils.isNetworkAvailable(getApplicationContext())) {
+            powerLantern.setChecked(false);
+            if (on) {
+                // User tried to turn Lantern on, but there's no 
+                // Internet connection available.
+                Utils.showAlertDialog(this, "Lantern", 
+                        getResources().getString(R.string.no_internet_connection));
+            } 
+            return;
         }
 
-        try {
-            int startTimeoutMillis = 60000;
-            String analyticsTrackingID = ""; // don't track analytics since those are already being tracked elsewhere
-            StartResult result = Lantern.enable(getApplicationContext(), startTimeoutMillis, analyticsTrackingID);
-            return result.getHTTPAddr();
-        }  catch (LanternNotRunningException lnre) {
-            throw new RuntimeException("Lantern failed to start: " + lnre.getMessage(), lnre);
-        }  
+        if (on) {
+            // Prompt the user to enable full-device VPN mode
+            // Make a VPN connection from the client
+            // We should only have one active VPN connection per client
+            try {
+                Log.d(TAG, "Load VPN configuration");
+                Intent intent = VpnService.prepare(getApplicationContext());
+                if (intent != null) {
+                    Log.w(TAG,"Requesting VPN connection");
+                    startActivityForResult(intent, REQUEST_VPN);
+                } else {
+                    Log.d(TAG, "VPN enabled, starting Lantern...");
+                    updateStatus(true);
+                    Lantern.disable(getApplicationContext());
```

Comment:
Yes, that's correct on both counts. The ``disable`` here is to make sure we clear the application proxy settings for the non-VPN instance. I can add a comment to clarify this.